File: rand.py
'''
Date: 2021.08.31 16:17:51
Description: Omit
LastEditors: Rustle Karl
LastEditTime: 2021.09.01 13:24:32
'''
from random import randint


def generate_lottery() -> tuple:
    red = set()

    while len(red) < 6:
        red.add(randint(1, 33))

    lottery = sorted(red)
    lottery.append(randint(1, 16))

    # 列表不可哈希
    return tuple(lottery)


def find_least_lottery() -> tuple:
    lotteries = set()

    while len(lotteries) < 1 << 12:
        lotteries.add(generate_lottery())

    while True:
        lottery = generate_lottery()
        if lottery not in lotteries:
            return lottery


if __name__ == '__main__':
    for _ in range(100):
        print(find_least_lottery())

File: rand.dart
/*
 * @Date: 2021.09.01 8:50
 * @Description: Omit
 * @LastEditors: Rustle Karl
 * @LastEditTime: 2021.09.01 8:50
 */
import 'dart:math';

List<int> generateLottery() {
  Random rand = Random();
  Set<int> red = {};
  while (red.length < 6) {
    red.add(rand.nextInt(32) + 1);
  }
  List<int> lottory = red.toList();
  lottory.sort();
  lottory.add(rand.nextInt(15) + 1);
  return lottory;
}

List<int> findLeastLottery() {
  Set<List<int>> lotteries = {};
  while (lotteries.length < 1 << 12) {
    lotteries.add(generateLottery()); // Dart
  }
  while (true) {
    List<int> lottery = generateLottery();
    if (!lotteries.contains(lottery)) {
      return lottery;
    }
  }
}
